py/dml/codegen.py

```diff
@@ -1260,10 +1260,19 @@ def try_resolve_len(site, lh):
 def expr_member(tree, location, scope):
     [lh, op, rh] = tree.args
     lh = codegen_expression_maybe_nonvalue(lh, location, scope)
-    if not tree.site.dml_version() == (1, 2) and op == '.' and rh == 'len':
-        member = try_resolve_len(tree.site, lh)
-        if member:
-            return member
+    if op == '.':
+        if not tree.site.dml_version() == (1, 2) and rh == 'len':
+            member = try_resolve_len(tree.site, lh)
+            if member:
+                return member
+        elif isinstance(lh, TemplatesRef):
```

Comment:
Perhaps `s/elif/if`, to cover this evil corner:
```
template len {
  method len() {}
}
group g is len;
...
g.templates.len.len();
```

Reply:
Also, this looks like a good opportunity to try the new `match` statement?

Reply:
I wondered "how" but then I found out that `isinstance` chains can be replaced by so-called "class patterns". Oh yes, that is a good idea.

Reply:
nevermind fisketur hates `match` :(

Reply:
Evil corner covered. Won't be using `match`.